src/storage/src/hummock/compaction_group_client.rs

```diff
@@ -16,47 +16,56 @@ use std::collections::HashMap;
 use std::sync::Arc;
 
 use itertools::Itertools;
+use parking_lot::RwLock;
 use risingwave_hummock_sdk::compaction_group::Prefix;
 use risingwave_hummock_sdk::CompactionGroupId;
 use risingwave_pb::hummock::CompactionGroup;
 use risingwave_rpc_client::HummockMetaClient;
-use tokio::sync::RwLock;
 
 use crate::hummock::{HummockError, HummockResult};
 
-/// `CompactionGroupClient` maintains compaction group metadata cache.
-pub struct CompactionGroupClient {
+#[async_trait::async_trait]
+pub trait CompactionGroupClient: Send + Sync + 'static {
+    async fn get_compaction_group_id(
+        &self,
+        prefix: Prefix,
+    ) -> HummockResult<Option<CompactionGroupId>>;
+}
+
+/// `CompactionGroupClientImpl` maintains compaction group metadata cache.
+pub struct CompactionGroupClientImpl {
     inner: RwLock<CompactionGroupClientInner>,
     hummock_meta_client: Arc<dyn HummockMetaClient>,
 }
 
-impl CompactionGroupClient {
+impl CompactionGroupClientImpl {
     pub fn new(hummock_meta_client: Arc<dyn HummockMetaClient>) -> Self {
         Self {
             inner: Default::default(),
             hummock_meta_client,
         }
     }
+}
 
-    /// Tries to get from local cache
-    pub async fn try_get_compaction_group_id(&self, prefix: Prefix) -> Option<CompactionGroupId> {
-        self.inner.read().await.get(&prefix)
-    }
-
-    /// Tries to get from meta service
-    pub async fn get_compaction_group_id(
+#[async_trait::async_trait]
+impl CompactionGroupClient for CompactionGroupClientImpl {
+    /// Tries to get from local cache first,then from meta service
+    async fn get_compaction_group_id(
```

Comment:
If shared buffer is already split by compaction group, I have a feeling that we will need to know more than just a compaction group id, for example, the compaction group configuration.

Reply:
i think so, hope that `CompactionGroupClient` can fetch more info when rpc, (link configuration) , in order to support ttl and so on

Reply:
The complete compaction group info is already given to CompactionGroupClient, but CompactionGroupClient only store part of it for now.
I think you can modify it as need later @Li0k .